### ISSUE TITLE
Make failing because of doctest::detail::MessageBuilder

### ISSUE THREAD

**Issue by MiguelMJ** [JustWhit3/osmanip#1]
When building with 
```
make
```
The following error occurs:
```
In file included from test/tests_helper_tools.cpp:3:
test/tests_helper_tools.cpp: In lambda function:
test/tests_helper_tools.cpp:15:32: error: no match for ‘operator+’ (operand types are ‘doctest::detail::MessageBuilder’ and ‘std::string’ {aka ‘std::__cxx11::basic_string<char>’})
   15 | #define test_string_hp "first" +                            \
      |                                ^
   16 |                        static_cast <std::string>(" \"") +   \
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                        |
      |                        std::string {aka std::__cxx11::basic_string<char>}
test/tests_helper_tools.cpp:65:80: note: in expansion of macro ‘test_string_hp’
   65 |   CHECK_THROWS_MESSAGE( throw( runtime_error_func( "first", var, "second" ) ), test_string_hp );
      |                                                                                ^~~~~~~~~~~~~~

```
As I'm not familiar with doctest, I would just suggest to separate the tests into their own build, with something like `make tests`.
Cheers!

**Comment by JustWhit3** [comment]
Hi @MiguelMJ , this issue has been solved with commit [f9bc091](https://github.com/JustWhit3/osmanip/commit/f9bc0915e25df6f5ecdcfdf636a729b80f454534).

**Comment by MiguelMJ** [comment]
It looks like there are other files causing the same error. I pulled the repo after the fix and still got this same error, in a different file:
```
test/tests_csmanip.cpp: In lambda function:
test/tests_csmanip.cpp:15:29: error: no match for ‘operator+’ (operand types are ‘doctest::detail::MessageBuilder’ and ‘std::string’ {aka ‘std::__cxx11::basic_string<char>’})
   15 | #define test_string "error" +                                      \
      |                             ^
   16 |                     static_cast <std::string>(" \"") +             \
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                     |
      |                     std::string {aka std::__cxx11::basic_string<char>}
test/tests_csmanip.cpp:49:45: note: in expansion of macro ‘test_string’
   49 |   CHECK_THROWS_MESSAGE( feat( sty, "not" ), test_string );
      |                                             ^~~~~~~~~~~

```

**Comment by JustWhit3** [comment]
Hi @MiguelMJ you are right. I forgot to fix the initialization of global variables also in the other test files. I fixed this in commit [2209ef1](https://github.com/JustWhit3/osmanip/commit/d1333ecc013f92882f7e0432b551bf7e27d1d5f8). Tell me if now it compiles correctly.

**Comment by MiguelMJ** [comment]
Fixed! It finally compiles directly. So nice!
